### PR TITLE
rules(no-disconnected): exclude compensation task and compensation boundary event

### PR DESCRIPTION
**Pull request by Nlea**
…ry event from the lint error

### CODE REVIEW

**Comment by nikku** [comment]
Thanks for your PR. Good catch on the compensation events.

CI is failing due to a couple of lint errors and we'd need to sort that out before merging. Did the following work for you locally?

```
npm run all
```

The lint rules specifically can be partly addressed automatically via the following command:

```
npm run lint -- --fix
```

File: rules/no-disconnected.js [hunk 1]
```diff
@@ -12,25 +12,39 @@ module.exports = function() {
 
   function check(node, reporter) {
 
-    if (!isAny(node, [
-      'bpmn:Task',
-      'bpmn:Gateway',
-      'bpmn:SubProcess',
-      'bpmn:Event'
-    ]) || node.triggeredByEvent) {
+    if (!isAny(node, ['bpmn:Task','bpmn:Gateway','bpmn:SubProcess','bpmn:Event']) || node.triggeredByEvent) {
```
Comment (nikku) [inline rules/no-disconnected.js:15]: Could you undo that re-format? I believe it is not necessary.

File: rules/no-disconnected.js [hunk 2]
```diff
@@ -12,25 +12,39 @@ module.exports = function() {
...
+      return;
+    }
+
+  // Check for compensation boundary events
+
+    if (node.$type == 'bpmn:BoundaryEvent'){
```
Comment (nikku) [inline rules/no-disconnected.js:21]: We have a `is` utility that we should use instead to check for a type:

```javascript
if (is(node, 'bpmn:BoundaryEvent')) {
   ...
}
```

File: rules/no-disconnected.js [hunk 3]
```diff
@@ -12,25 +12,39 @@ module.exports = function() {
...
+     var eventDefinitions = node.get('eventDefinitions');      
+      if (eventDefinitions.length == 1){
```
Comment (nikku) [inline rules/no-disconnected.js:23]: Use `===` for strict comparison (no magic type casting) in JavaScript.

File: rules/no-disconnected.js [hunk 4]
```diff
@@ -12,25 +12,39 @@ module.exports = function() {
...
+          if (node.eventDefinitions[0].$type == 'bpmn:CompensateEventDefinition'){
```
Comment (nikku) [inline rules/no-disconnected.js:24]: Re-use existing `eventDefinitions` variable here?

File: test/rules/no-disconnected.js [hunk 5]
```diff
@@ -17,7 +17,12 @@ RuleTester.verify('no-disconnected', rule, {
     },
     {
       moddleElement: readModdle(__dirname + '/no-disconnected/valid-event-subprocess.bpmn')
+    },
+    {
+      moddleElement: readModdle(__dirname + '/no-disconnected/valid-compensation.bpmn')
```
Comment (nikku) [inline test/rules/no-disconnected.js:22]: Great you added a test case, too :+1:

**Review by nikku — CHANGES_REQUESTED** [review]
Some minor code style improvements.

To semi-automatically fix some of the code formatting issues, run `npm run lint -- --fix`.

File: package.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
   "description": "Validate your BPMN diagrams based on configurable lint rules",
   "version": "7.2.0",
   "main": "lib/index.js",
+  "module": "lib/main.js",
```
Comment (nikku) [inline package.json:6]: What exactly are these changes about here?

Comment (Nlea) [reply package.json:6]: I did not change them actively... maybe my IDE did?

Comment (nikku) [reply package.json:6]: This is due to the merge you did.

Make sure your feature branch is up-to-date with the upstream (bpmnlint master):

```
git rebase upstream/master
```

**Comment by nikku** [comment]
I've cleaned up your PR via https://github.com/bpmn-io/bpmnlint/pull/60.

**Comment by nikku** [comment]
What is left is the check whether the compensation association exists. I won't pursue that one for now though.

**Comment by nikku** [comment]
Merged via https://github.com/bpmn-io/bpmnlint/commit/54f181291952d12d5a1f7d938346a10fac1d41e2.

**Comment by nikku** [comment]
Released as `v7.2.1`.

Pending integration in downstream libraries and Camunda Modeler Linter Plug-in.

**Pull request closed without merging**